### ISSUE TITLE
Is writing "image-label" supported?

### ISSUE THREAD

**Issue by constantinpape** [ome/ome-zarr-py#171]
Is there support to write "image-label" datasets (https://ngff.openmicroscopy.org/latest/#label-md) in ome-zarr-py already?
If yes, how do I use it?

**Comment by will-moore** [comment]
Threre's an example at https://github.com/ome/ome-zarr-py/blob/0ea5bf61ef754069abf4f223897d55630c953fbf/README.rst
which is part of a long-standing PR https://github.com/ome/ome-zarr-py/pull/121
Actually, I have subsequently removed the labels from the PR to make it more concise for the README

**Comment by constantinpape** [comment]
Thanks @will-moore; the example was very helpful and I could update my scripts to write ome.zarr with labels.

**Comment by sbesson** [comment]
Looking at the example, I could imagine wrapping the succession of low-level calls into utilities dealing with the group/metadata creation/update e.g.

```
write_image(image=data, group=image_root)
label_root = add_label(image_root, label_name)
write_image(image=label_data, group=label_root)
```

or 

```
write_image(image=data, group=image_root)
write_label(image=image_root, group=image_root, name=label_name)
```

It would be interesting to know you would see the value of introducing such API based on your usage @constantinpape.

**Comment by constantinpape** [comment]
> It would be interesting to know you would see the value of introducing such API based on your usage @constantinpape.

Yes, absolutely! `write_label` should also have support for (optional? depending on https://github.com/ome/ngff/issues/105) arguments to pass the label/property values.

**Comment by sbesson** [comment]
I have updated the title of this issue to reflect the new scope.

So current preference would be rather a single API that would first create/update the intermediate groups (depending on the presence of labels) and then write the label image, is that correct? In terms of behavior, I agree the API should also support both mandatory as well as optional `image-label` attributes.

For the metadata, my current preference would be to extend `write_multiscale` to support an extra metadata dictionary similarly `storage_options`. This would allow to set additional metadata, either specified (`omero`, `image-label`...) or custom, to be stored in the attributes of the multiscales group.

Thinking of the sequence of a `write_label_image` API, the biggest unknown to me at this point is whether this API should call `write_image` or `write_multiscales`. The former is effectively a wrapper around the latter but I do not know whether it should always be seen as the single entry-point @joshmoore @will-moore ?

**Comment by constantinpape** [comment]
Implemented in #178
